### ISSUE TITLE
Download failed on Cask 'telegram' for version 3.8-118923

### ISSUE THREAD

**Issue by nico-arianto** [caskroom/homebrew-cask#44673]
#### General troubleshooting steps

- [x] I have retried my command with `--force` and the issue is still present.
- [x] I have checked the instructions for [reporting bugs](https://github.com/caskroom/homebrew-cask#reporting-bugs) (or [making requests](https://github.com/caskroom/homebrew-cask#requests)) before opening the issue.
  - [x] None of the templates was appropriate for my issue, or I’m not sure.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/caskroom/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

#### Description of issue

Download failed on Cask 'telegram' with message: Download failed: https://osx.telegram.org/updates/Telegram-3.8-118923.app.zip

#### Output of your command with `--verbose --debug`

```
/usr/bin/curl --show-error --user-agent Homebrew/1.5.9 (Macintosh; Intel Mac OS X 10.13.3) curl/7.54.0 --fail --location --remote-time --continue-at - --output /Users/nicoarianto/Library/Caches/Homebrew/Cask/telegram--3.8-118923.zip.incomplete https://osx.telegram.org/updates/Telegram-3.8-118923.app.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
Error: Download failed on Cask 'telegram' with message: Download failed: https://osx.telegram.org/updates/Telegram-3.8-118923.app.zip
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/download.rb:42:in `rescue in fetch'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/download.rb:40:in `fetch'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/download.rb:15:in `perform'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:126:in `download'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:51:in `fetch'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:75:in `install'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/installer.rb:104:in `reinstall'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/reinstall.rb:10:in `block in run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/reinstall.rb:5:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/reinstall.rb:5:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli/abstract_command.rb:35:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:98:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:168:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:132:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:8:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
Error: Kernel.exit
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:173:in `exit'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:173:in `rescue in run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:156:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/lib/hbc/cli.rb:132:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:8:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
```

#### Output of `brew cask doctor`

```
==> Homebrew-Cask Version
Homebrew-Cask 1.5.9
caskroom/homebrew-cask (git revision bdb8f; last commit 2018-03-08)
==> macOS
10.13.3
==> SIP
Enabled
==> Java
1.8.0_162
==> Homebrew-Cask Install Location
<NONE>
==> Homebrew-Cask Staging Location
/usr/local/Caskroom
==> Homebrew-Cask Cached Downloads
~/Library/Caches/Homebrew/Cask
==> Homebrew-Cask Taps:
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask (3928 casks)
/usr/local/Homebrew/Library/Taps/caskroom/homebrew-versions (173 casks)
==> Contents of $LOAD_PATH
/usr/local/Homebrew/Library/Homebrew/cask/lib
/usr/local/Homebrew/Library/Homebrew
/Library/Ruby/Gems/2.3.0/gems/did_you_mean-1.0.0/lib
/Library/Ruby/Site/2.3.0
/Library/Ruby/Site/2.3.0/x86_64-darwin17
/Library/Ruby/Site/2.3.0/universal-darwin17
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/universal-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/x86_64-darwin17
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/universal-darwin17
==> Environment Variables
LC_ALL="en_US.UTF-8"
PATH="/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/Homebrew/Library/Homebrew/shims/scm"
SHELL="/bin/bash"
```

**Comment by nico-arianto** [comment]
Telegram app in App Store still version `3.7.5`.

**Comment by vitorgalvao** [comment]
https://github.com/caskroom/homebrew-cask/blob/master/doc/reporting_bugs/a_cask_fails_to_install.md#curl-error

https://github.com/caskroom/homebrew-cask/pull/44670

**Comment by vid-uo** [comment]
Hi @vitorgalvao

I think this issue persists. The cask points at Telegram version: 3.8-118923 and that zip is no longer available for download. Following the cask code: I checked the versions.xml file and I see a new version: 3.8-118941 which was released today (March 8th). 
I suspect Telegram removed version 3.8-118923 from the downloads.

@vbaluch just created the pull request updating the cask to 3.8-118923 yesterday; approved and merged today [#44641](https://github.com/caskroom/homebrew-cask/pull/44641).

It passed the tests before it was merged but then Telegram released a new version.

Here's the audit output at the moment:

```
$ brew cask audit --download telegram
==> Downloading https://osx.telegram.org/updates/Telegram-3.8-118923.app.zip

curl: (22) The requested URL returned error: 404 Not Found
audit for telegram: failed
 - download not possible: Download failed on Cask 'telegram' with message: Download failed: https://osx.telegram.org/updates/Telegram-3.8-118923.app.zip
 - appcast checkpoint mismatch
Expected: 0d10c69eb35bc6d29600522330debeda38d898923f981ba8b50bf2a98bb72f78
Actual: 72c90006eaddeec64dc93de817846235add24ccb6cb38d59cda12f8fd7012709
Error: audit failed for casks: telegram
```
